src/main/java/com/aws/greengrass/ipc/GetClientDeviceAuthTokenOperationHandler.java

```diff
@@ -0,0 +1,134 @@
+import static software.amazon.awssdk.aws.greengrass.GreengrassCoreIPCService.GET_CLIENT_DEVICE_AUTH_TOKEN;
+
+public class GetClientDeviceAuthTokenOperationHandler
+        extends GeneratedAbstractGetClientDeviceAuthTokenOperationHandler {
+    private static final Logger logger = LogManager.getLogger(GetClientDeviceAuthTokenOperationHandler.class);
+    private static final String COMPONENT_NAME = "componentName";
+    private static final String UNAUTHORIZED_ERROR = "Not Authorized";
+    private static final String MQTT_CREDENTIAL_TYPE = "mqtt";
+    private static final String NO_DEVICE_CREDENTIAL_ERROR = "Client device credential document is required";
+    private static final String NO_VALID_MQTT_CREDENTIAL_ERROR = "Valid mqtt credential document is required";
+    private final String serviceName;
+    private final AuthorizationHandler authorizationHandler;
+    private final SessionManager sessionManager;
+    private final Map<String, String> credentialMap = new HashMap<>();
+
+    /**
+     * Constructor.
+     *
+     * @param context              operation continuation handler
+     * @param sessionManager       session manager
+     * @param authorizationHandler authorization handler
+     */
+    public GetClientDeviceAuthTokenOperationHandler(
+            OperationContinuationHandlerContext context,
+            SessionManager sessionManager,
+            AuthorizationHandler authorizationHandler
+    ) {
+
+        super(context);
+        serviceName = context.getAuthenticationData().getIdentityLabel();
+        this.sessionManager = sessionManager;
+        this.authorizationHandler = authorizationHandler;
+    }
+
+    @SuppressWarnings({"PMD.AvoidCatchingGenericException", "PMD.PreserveStackTrace"})
+    @Override
+    public GetClientDeviceAuthTokenResponse handleRequest(GetClientDeviceAuthTokenRequest request) {
+        return translateExceptions(() -> {
+            try {
+                doAuthorizationForClientDevAuthToken();
+            } catch (AuthorizationException e) {
+                logger.atWarn().kv("error", e.getMessage()).kv(COMPONENT_NAME, serviceName).log(UNAUTHORIZED_ERROR);
+                throw new UnauthorizedError(e.getMessage());
+            }
+            Map<String, String> credentialMap = mapOfMqttCredential(request.getCredential());
+            try {
+                String sessionId = sessionManager.createSession(MQTT_CREDENTIAL_TYPE, credentialMap);
+                GetClientDeviceAuthTokenResponse response = new GetClientDeviceAuthTokenResponse();
+                return response.withClientDeviceAuthToken(sessionId);
+            } catch (AuthenticationException e) {
+                logger.atError().kv("error", e.getMessage()).kv(COMPONENT_NAME, serviceName)
+                        .log("Unable to authenticate the client device with the given credentials");
+                throw new InvalidCredentialError(e.getMessage());
```

Comment:
Don't include the exception message. We don't want to return that to the client.